numba/tests/test_analysis.py

```diff
@@ -246,8 +246,6 @@ def impl(x):
             x_is_none_work = False
             if x is None:
                 x_is_none_work = True
-            else:
-                pass  # force the True branch exit to be on backbone
 
```

Comment:
I'm not sure that this change is valid. Am looking into it.

Reply:
it originates from `https://github.com/numba/numba/pull/7381/commits/036d43b778a8dd16791ee4b8cc893c5bdc2f5883

Reply:
Try this:
```diff
diff --git a/numba/tests/test_analysis.py b/numba/tests/test_analysis.py
index f89042f..3a75019 100644
--- a/numba/tests/test_analysis.py
+++ b/numba/tests/test_analysis.py
@@ -246,6 +246,8 @@ class TestBranchPrune(TestBranchPruneBase, SerialMixin):
             x_is_none_work = False
             if x is None:
                 x_is_none_work = True
+            else:
+                pass
 
             if x_is_none_work:
                 y = 10
@@ -253,7 +255,14 @@ class TestBranchPrune(TestBranchPruneBase, SerialMixin):
                 y = -3
             return y
 
-        self.assert_prune(impl, (types.NoneType('none'),), [None, None], None)
+        if utils.PYVERSION >= (3, 10):
+            # Python 3.10 creates a block with a NOP in it for the `pass` which
+            # means it gets pruned.
+            self.assert_prune(impl, (types.NoneType('none'),), [False, None],
+                              None)
+        else:
+            self.assert_prune(impl, (types.NoneType('none'),), [None, None],
+                              None)
         self.assert_prune(impl, (types.IntegerLiteral(10),), [True, None], 10)
 ```

Reply:
I don't think there's a way to create the scenario that was originally being tested in Python 3.10, but there's value in having the test for `< 3.10`. In Python 3.10 the `else: pass` ends up as a block with a `NOP` opposed to a direct jump from the `if` to the backbone, the above adjusts the checking for the expected behaviours.

Reply:
applied as 3edd2d79f51c68c99e3dae5fe470d54473cdcd3f

Reply:
Is this patch (3edd2d7) on this branch?

Reply:
now as a5497934cc3faa8d46e12b97d59f166faf5e4161

Reply:
now as d7558eb88bc01428c63679124f775ca02a3a593e